pkg/volume/cinder/attacher.go

```diff
@@ -124,7 +124,18 @@ func (attacher *cinderDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nod
 		volumesAttachedCheck[spec] = true
 		volumeSpecMap[volumeSource.VolumeID] = spec
 	}
-	attachedResult, err := attacher.cinderProvider.DisksAreAttached(volumeIDList, string(nodeName))
+	instances, res := attacher.cinderProvider.Instances()
+	if !res {
+		return volumesAttachedCheck, fmt.Errorf("failed to list openstack instances")
```

Comment:
`Instances()` is a call to get the instance api from cloud provider, the error message should be more appropriate to be `the underlying cloud provider does not claim to support Instances api`.